src/community/ogcapi/ogcapi-coverages/src/test/java/org/geoserver/ogcapi/coverages/CollectionsTest.java

```diff
@@ -0,0 +1,270 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.ogcapi.coverages;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.jayway.jsonpath.DocumentContext;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import net.minidev.json.JSONArray;
+import org.geoserver.catalog.CoverageInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.config.GeoServerInfo;
+import org.geoserver.config.SettingsInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.wcs.WCSInfo;
+import org.hamcrest.Matchers;
+import org.jsoup.Jsoup;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+public class CollectionsTest extends CoveragesTestSupport {
+
+    public static final String TAZDEM_TITLE = "Down under digital elevation model";
+    public static final String TAZDEM_DESCRIPTION = "So that you know where up and down are";
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+
+        // customize metadata and set custom CRS too
+        CoverageInfo tazDem = getCatalog().getCoverageByName("rs:DEM");
+        tazDem.setTitle(TAZDEM_TITLE);
+        tazDem.setAbstract(TAZDEM_DESCRIPTION);
+        tazDem.getResponseSRS().addAll(Arrays.asList("3857", "3003"));
+        getCatalog().save(tazDem);
+    }
+
+    @Test
+    public void testCollectionsJson() throws Exception {
+        DocumentContext json = getAsJSONPath("ogc/coverages/collections", 200);
+        testCollectionsJson(json);
+    }
+
+    private void testCollectionsJson(DocumentContext json) throws Exception {
+        int expected = getCatalog().getCoverages().size();
+        assertEquals(expected, (int) json.read("collections.length()", Integer.class));
+
+        //        // check we have the expected number of links and they all use the right "rel"
```

Comment:
Could probably just junk all of this commented code.

Reply:
Actually I commented out while I was waiting to have the coverage links per output format, and then forgot to uncomment. Fixing.